vue/components/SharedSubComponents/infiniteLoaderMixin.js

```diff
@@ -0,0 +1,90 @@
+const scrollThreshold = 400
+
+// infinite-loader-mixin: add to your component, call startLoading(string baseURL, boolean useHistory) to (re)start it as many times as you want, and access
+// the results through the results array and access the loading state, if you want to display it, through loading (boolean) and allResultsLoaded (boolean).
+// overwrite the getNextRequestURL(string baseURL) method to use non-oldest-result-loaded-timestamp-based pagination methods.
+// Accessing or modifying the state through any other means is done completely at your own risk.
+export default {
+  data: function () {
+    return {
+      loading: false,
+      oldestResultLoaded: new Date().getTime(),
```

Comment:
Using timestamps for this is fine, but I suspect we could use ETags or similar purpose-driven elements of HTTP to clean this up, in future. I'll need to read up on this, though, as it's something I've only glancingly encountered.